packages/flutter/lib/src/material/outline_button.dart

```diff
@@ -339,37 +356,46 @@ class _OutlineButtonState extends State<OutlineButton> with SingleTickerProvider
 
   // TODO(hmuller): this method is the same as FlatButton
   Color _getSplashColor(ThemeData theme, ButtonThemeData buttonTheme) {
-    if (widget.splashColor != null)
-      return widget.splashColor;
+    if (widget.splashColor != null) return widget.splashColor;
 
     switch (_getTextTheme(buttonTheme)) {
       case ButtonTextTheme.normal:
       case ButtonTextTheme.accent:
         return theme.splashColor;
       case ButtonTextTheme.primary:
         return theme.brightness == Brightness.dark
-          ? Colors.white12
-          : theme.primaryColor.withOpacity(0.12);
+            ? Colors.white12
+            : theme.primaryColor.withOpacity(0.12);
     }
     return Colors.transparent;
   }
 
+  /// The button outline has a color and a width.
+  ///
+  /// If [this.borderSize] is provided and the style is none, then that is used.
+  ///
+  /// Else, the color and width resolve to the most specific provided value,
+  /// or the first available theme value.
   BorderSide _getOutline(ThemeData theme, ButtonThemeData buttonTheme) {
-    final bool themeIsDark = theme.brightness == Brightness.dark;
-    if (widget.borderSide?.style == BorderStyle.none)
-      return widget.borderSide;
+    if (widget.borderSide?.style == BorderStyle.none) return widget.borderSide;
+
+    Color color = widget.borderSide?.color;
+    if (_pressed) {
+      color = widget.highlightedBorderColor ?? color;
+    }
+    if (!widget.enabled) {
+      color = widget.disabledBorderColor ?? color;
+    }
```

Comment:
the second if here stomps on the other one. Either chain them with `else`, or reverse their order, then chain them with `else`, depending on which order you want.

Reply:
oh, i see, this is just updating color over and over but defaulting to the previous value. that's confusing.

Reply:
we should pick a way to do this here, and stick to it. Either use `??=` all the way down, so that you list the colors in preference order, or use `?? color` all the way down, so that you list them in reverse preference order.

The former is probably better for performance.

You could go even further and nest all the conditionals so that as soon as you've found the right color, you bail with the winning color.

Reply:
Reverted back to original nesting format. I did I find it difficult to parse with nested ternaries and ?? operators together, but perhaps that is because the spacing was off. I updated the spacing to reflect the style guides, but am still not sure if its correct style wise. PTAL.